test/utils/shoots/update/inplace/inplace.go

```diff
@@ -0,0 +1,140 @@
+			if hasAutoInplaceUpdate {
+				g.Expect(shoot.Status.InPlaceUpdates.PendingWorkerUpdates.AutoInPlaceUpdate).NotTo(BeEmpty())
+			}
+			if hasManualInplaceUpdate {
+				g.Expect(shoot.Status.InPlaceUpdates.PendingWorkerUpdates.ManualInPlaceUpdate).NotTo(BeEmpty())
+				g.Expect(shoot.Status.Constraints).To(ContainCondition(
+					OfType(gardencorev1beta1.ShootManualInPlaceWorkersUpdated),
+					WithReason("WorkerPoolsWithManualInPlaceUpdateStrategyPending"),
+					Or(WithStatus(gardencorev1beta1.ConditionFalse), WithStatus(gardencorev1beta1.ConditionProgressing)),
+				))
+			}
+		}).Should(Succeed())
+	}, SpecTimeout(2*time.Minute))
+}
+
+// LabelManualInPlaceNodesWithSelectedForUpdate labels all manual in-place nodes with the selected-for-update label.
+// In the actual scenario, this should be done by the user, but for testing purposes, we do it here.
+func LabelManualInPlaceNodesWithSelectedForUpdate(ctx context.Context, shootClient client.Client, shoot *gardencorev1beta1.Shoot) {
+	GinkgoHelper()
+
+	for _, pool := range shoot.Spec.Provider.Workers {
+		if !v1beta1helper.IsUpdateStrategyManualInPlace(pool.UpdateStrategy) {
+			continue
+		}
+		nodeList := &corev1.NodeList{}
+		Eventually(ctx, func(g Gomega) {
+			g.Expect(shootClient.List(ctx, nodeList, client.MatchingLabels{v1beta1constants.LabelWorkerPool: pool.Name})).To(Succeed())
+		}).Should(Succeed())
+
+		for _, node := range nodeList.Items {
+			if metav1.HasLabel(node.ObjectMeta, machinev1alpha1.LabelKeyNodeSelectedForUpdate) {
+				continue
+			}
+
+			metav1.SetMetaDataLabel(&node.ObjectMeta, machinev1alpha1.LabelKeyNodeSelectedForUpdate, "true")
+			Eventually(ctx, func(g Gomega) {
+				g.Expect(shootClient.Update(ctx, &node)).To(Succeed())
+			}).Should(Succeed(), "node %s should be labeled", node.Name)
+		}
+	}
+}
+
+// FindAllNodesOfInPlaceWorker finds all nodes of in-place workers and returns their names.
+func FindAllNodesOfInPlaceWorker(ctx context.Context, shootClient client.Client, shoot *gardencorev1beta1.Shoot) sets.Set[string] {
+	GinkgoHelper()
+
+	nodesOfInPlaceWorkers := sets.New[string]()
+
+	for _, pool := range shoot.Spec.Provider.Workers {
+		if !v1beta1helper.IsUpdateStrategyInPlace(pool.UpdateStrategy) {
+			continue
+		}
+		nodeList := &corev1.NodeList{}
+		Eventually(ctx, func(g Gomega) {
+			g.Expect(shootClient.List(context.Background(), nodeList, client.MatchingLabels{v1beta1constants.LabelWorkerPool: pool.Name})).To(Succeed())
+		}).Should(Succeed())
+
+		for _, node := range nodeList.Items {
+			nodesOfInPlaceWorkers.Insert(node.Name)
+		}
+	}
+
+	return nodesOfInPlaceWorkers
+}
+
+// ItShouldVerifyInPlaceUpdateCompletion verifies that the in-place update was completed successfully by checking the
+// .status.inPlaceUpdates and the ManualInPlaceWorkersUpdated constraint of the Shoot.
+func ItShouldVerifyInPlaceUpdateCompletion(gardenClient client.Client, shoot *gardencorev1beta1.Shoot) {
```

Comment:
Can move `ItShouldVerifyInPlaceUpdateStart` and this function to a file called `shoot.go` and rename this files to `nodes.go`